test/integration/framework/master_utils.go

```diff
@@ -0,0 +1,318 @@
+func StartRC(controller *api.ReplicationController, restClient *client.Client) (*api.ReplicationController, error) {
+	created, err := restClient.ReplicationControllers(controller.Namespace).Get(controller.Name)
+	if err != nil {
+		glog.Infof("Rc %v doesn't exist, creating", controller.Name)
+		created, err = restClient.ReplicationControllers(controller.Namespace).Create(controller)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// If we just created an rc, wait till it creates its replicas.
+	return ResizeRC(created.Name, created.Namespace, controller.Spec.Replicas, restClient)
+}
+
+// StartPods check for numPods in TestNS. If they exist, it no-ops, otherwise it starts up
+// a temp rc, resizes it to match numPods, then deletes the rc leaving behind the pods.
+func StartPods(numPods int, host string, restClient *client.Client) error {
+	start := time.Now()
+	defer func() {
+		glog.Infof("StartPods took %v with numPods %d", time.Since(start), numPods)
+	}()
+	hostField := fields.OneTermEqualSelector(client.PodHost, host)
+	pods, err := restClient.Pods(TestNS).List(labels.Everything(), hostField)
+	if err != nil || len(pods.Items) == numPods {
+		return err
+	}
+	glog.Infof("Found %d pods that match host %v, require %d", len(pods.Items), hostField, numPods)
+	// For the sake of simplicity, assume all pods in TestNS have selectors matching TestRCManifest.
+	controller := RCFromManifest(TestRCManifest)
+
+	// Make the rc unique to the given host.
+	controller.Spec.Replicas = numPods
+	controller.Spec.Template.Spec.Host = host
+	controller.Name = controller.Name + host
+	controller.Spec.Selector["host"] = host
+	controller.Spec.Template.Labels["host"] = host
+
+	if rc, err := StartRC(controller, restClient); err != nil {
+		return err
+	} else {
+		// Delete the rc, otherwise when we restart master components for the next benchmark
+		// the rc controller will race with the pods controller in the rc manager.
+		return restClient.ReplicationControllers(TestNS).Delete(rc.Name)
+	}
+}
+
+// TODO: Merge this into startMasterOrDie.
+func RunAMaster(t *testing.T) (*master.Master, *httptest.Server) {
+	helper, err := master.NewEtcdHelper(NewEtcdClient(), testapi.Version(), etcdtest.PathPrefix())
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	m := master.New(&master.Config{
+		EtcdHelper:        helper,
+		KubeletClient:     client.FakeKubeletClient{},
+		EnableLogsSupport: false,
+		EnableProfiling:   true,
+		EnableUISupport:   false,
+		APIPrefix:         "/api",
+		Authorizer:        apiserver.NewAlwaysAllowAuthorizer(),
+		AdmissionControl:  admit.NewAlwaysAdmit(),
+	})
+
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		m.Handler.ServeHTTP(w, req)
+	}))
+
+	return m, s
+}
+
+// Task is a function passed to worker goroutines by RunParallel.
+// The function needs to implement its own thread safety.
+type Task func(id int) error
```

Comment:
figured the id would be useful when testing watch/update, since I need some way to get a worker to focus on a subset of a list of pods or their updates will all compete with each other.